ToolChain/Assembler/Parser/Parser.py

```diff
@@ -162,239 +179,128 @@ def _parseCodeLine(self, line: list=None):
-        """
-        Will translate a text FLAGs to flags code as:
-        FLAGS: 0b000 : Zero, Lower, Higher
-        :param textFlags:
-        :return:
-        """
-        codeFlags = 0b000
-        originalFlags = textFlags
-        textFlags = textFlags.lower()
-
-        if "z" in textFlags or "e" in textFlags:
-            codeFlags |= 0b100
-            textFlags = textFlags.replace("z", "")
-            textFlags = textFlags.replace("e", "")
-
-        if "l" in textFlags:
-            codeFlags |= 0b010
-            textFlags = textFlags.replace("l", "")
-
-        if "h" in textFlags:
-            codeFlags |= 0b001
-            textFlags = textFlags.replace("h", "")
-
-        if len(textFlags) > 0:
-            # Invalid flag selection detected!
-            raise ValueError("Invalid conditional flag detected {} was provided but is invalid".format(originalFlags))
-
-        return codeFlags
-
     def _evaluateAndExtractInfoFromLine(self, line, buildInstruction):
         """
-        This is an helper method. The aim to this is to keep caller code cleaner
-        TODO: refactor this... it's ugly
+        This is a helper method. The aim to this is to keep caller code cleaner
+        Method evaluates each part of the line to determine whether we're dealing with an instruction, label, data, etc.
+        We begin by verifying if the first argument has label or data mnemonics (not an instruction).
+        If none are found, we assume it's an instruction. We then proceed to evaluating the arguments.
+        The method uses the Instruction class' source and destination fields to enforce proper instruction format.
+        For example, if the source and destination have already been found, additional arguments means the line is an
+        invalid instruction.
+
         :param line:
         :param buildInstruction:
         :return:
         """
-        foundSource = False  # Found source operand
-        foundDestination = False  # Keep us from abusive operation
-
-        if (MEMORY_REFERENCE_INDICATORS == line[0][-1]) or COMMENT_INDICATORS == line[0][0]:
-            # This is no code line... No need for further work, return!
+        foundSource = False                              # Flag for whether we found a source immediate or register
+        foundDestination = False                         # Flag for whether we found a destination immediate or register
+        indicator_list = [COMMENT_INDICATORS,            # List of non-instruction indicators
+                          MEMORY_REFERENCE_INDICATORS,
+                          DATA_ALPHA_INDICATOR,
+                          DATA_NUMERIC_INDICATOR,
+                          DATA_MEMORY_REFERENCE,
+                          EXPORTED_REFERENCE_INDICATOR]
+
+        sline = line.split()
+        buildInstruction.operationMnemonic = sline[0].upper()
+
+        # PART 1: Evaluate the first part of the line, which should be an instruction or mnemonic
```

Comment:
Actually, the first part looks for items that aren't instructions such as labels (mem refs) and handles them differently.